Modules/Devices/src/Devices.Infrastructure/PushNotifications/FirebaseCloudMessaging/FirebaseCloudMessagingConnector.cs

```diff
@@ -1,44 +1,43 @@
-﻿using System.Text.Json;
+﻿using System.Collections.Immutable;
+using System.Text.Json;
 using Backbone.Modules.Devices.Application.Extensions;
 using Backbone.Modules.Devices.Application.Infrastructure.PushNotifications;
 using Backbone.Modules.Devices.Infrastructure.PushNotifications.AzureNotificationHub;
 using Backbone.Modules.Devices.Infrastructure.PushNotifications.DirectPush;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
 using FirebaseAdmin.Messaging;
 
 namespace Backbone.Modules.Devices.Infrastructure.PushNotifications.FirebaseCloudMessaging;
 public class FirebaseCloudMessagingConnector : IPnsConnector
 {
-    public FirebaseCloudMessagingConnector()
-    { }
+    private readonly FirebaseMessaging _firebaseMessaging;
 
-    public async Task Send(IEnumerable<PnsRegistration> registrations, object notification)
+    public FirebaseCloudMessagingConnector(FirebaseMessaging firebaseMessaging)
+    {
+        _firebaseMessaging = firebaseMessaging;
+    }
+
+    public async Task Send(IEnumerable<PnsRegistration> registrations, IdentityAddress recipient,
```

Comment:
I had considered this. But then I think that it could be possible that some weird shenanigans might occur, like a bunch of people registering the same identity on a lot of devices. I will remove it nonetheless.

Reply:
Yeah, we should rather add some validation for that case, so that you can register more than n devices. But for now I'm okay with it to fail in case someone has that many devices.